tests/testsuite/doc.rs

```diff
@@ -1638,3 +1639,145 @@ fn crate_versions_flag_is_overridden() {
     p.cargo("rustdoc -- --crate-version 2.0.3").run();
     asserts(output_documentation());
 }
+
+#[cargo_test]
+fn doc_fingerprint_versioning_consistent() {
+    #[derive(Debug, Serialize, Deserialize)]
+    pub struct RustDocFingerprint {
+        pub rustc_verbose_version: String,
+    }
+
+    // Test that using different Rustc versions forces a
+    // doc re-compilation producing new html, css & js files.
+
+    // Random rustc verbose version
+    let stable_rustc_verbose_version = format!(
+        "\
+rustc 1.41.1 (f3e1a954d 2020-02-24)
+binary: rustc
+commit-hash: f3e1a954d2ead4e2fc197c7da7d71e6c61bad196
+commit-date: 2020-02-24
+host: {}
+release: 1.41.1
+LLVM version: 9.0
+",
+        rustc_host()
+    );
+
+    // Create the dummy project.
+    let dummy_project = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "foo"
+            version = "1.2.4"
+            authors = []
+        "#,
+        )
+        .file("src/lib.rs", "//! These are the docs!")
+        .build();
+
+    dummy_project.cargo("doc").run();
+
+    let fingerprint: RustDocFingerprint = serde_json::from_str(
+        dummy_project
+            .read_file(
+                std::path::PathBuf::from("target")
+                    .join("doc")
+                    .join(".rustdoc_fingerprint.json")
+                    .to_str()
+                    .expect("Malformed path"),
+            )
+            .as_str(),
+    )
+    .expect("JSON Serde fail");
```

Comment:
There's no need to create a path here, something like the following should be sufficient.

```rust
    let fingerprint: RustDocFingerprint =
        serde_json::from_str(&dummy_project.read_file("target/doc/.rustdoc_fingerprint.json"))
            .expect("JSON Serde fail");
```